packages/atlas/src/views/studio/PlaylistWorkspace/VideoSelectorDialog/VideoSelectorDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from 'react'
+
+import { useFullVideosConnection } from '@/api/hooks/videosConnection'
+import { VideoOrderByInput } from '@/api/queries/__generated__/baseTypes.generated'
+import { SvgActionSearch } from '@/assets/icons'
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { Input } from '@/components/_inputs/Input'
+import { DialogModal } from '@/components/_overlays/DialogModal'
+import { VideoListItem, VideoListItemLoader } from '@/components/_video/VideoListItem'
+import { useDebounceValue } from '@/hooks/useDebounceValue'
+import { useAuthorizedUser } from '@/providers/user/user.hooks'
+import { SentryLogger } from '@/utils/logs'
+
+import { Wrapper } from './VideoSelectorDialog.styles'
+
+type VideoSelectorDialogProps = {
+  onHide: () => void
+  show: boolean
+  onSelect: (arg: string[]) => void
+  initialSelected?: string[]
+}
+const INITIAL_FIRST = 50
+
+export const VideoSelectorDialog = ({ onHide, show, onSelect, initialSelected = [] }: VideoSelectorDialogProps) => {
+  const [selectedVideos, setSelectedVideos] = useState<string[]>(initialSelected)
+  const [search, setSearch] = useState<string>('')
+  const { channelId } = useAuthorizedUser()
+  const debouncedSearch = useDebounceValue(search, 200)
+  const { edges, loading } = useFullVideosConnection(
+    {
+      first: INITIAL_FIRST,
+      orderBy: VideoOrderByInput.CreatedAtDesc,
+      where: {
+        channel: {
+          id_eq: channelId,
+        },
+        title_contains: debouncedSearch,
+      },
+    },
+    {
+      notifyOnNetworkStatusChange: true,
+      onError: (error) => SentryLogger.error('Failed to fetch videos', 'MyVideosView', error),
+    }
+  )
+
+  useEffect(() => {
+    if (show) {
+      setSelectedVideos(initialSelected)
+    }
+    // need to update selected videos when modal opens
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm a little worried that it could cause bugs. What if `initalSelected` change?

Reply:
Do we actually need this effect?

Reply:
It depends on what behavior we want to have (eventually I can introduce some callback to track videos when the modal is closed). If modal should remember what videos are selected, then yes. If we want the modal to always show videos as unselected at the beginning then it can be removed. WDYT

Reply:
> I'm a little worried that it could cause bugs. What if `initalSelected` change?

We only want to update the state when modal reopens, since it's mounted all the time I don't see another cleaner option